lib/tests/it/extract.rs

```diff
@@ -0,0 +1,228 @@
+use async_tempfile::TempDir;
+use circe_lib::{
+    extract::{extract, Report, Strategy},
+    registry::Registry,
+    Digest, Reference,
+};
+use color_eyre::Result;
+use serde_json::{json, Value};
+use simple_test_case::test_case;
+use std::{path::PathBuf, str::FromStr};
+
+macro_rules! assert_layers_extracted {
+    ($report:expr, $layers:expr) => {
+        pretty_assertions::assert_eq!(
+            $report
+                .layers
+                .iter()
+                .map(|(l, _)| l.to_string())
+                .collect::<Vec<_>>(),
+            $layers.into_iter().collect::<Vec<_>>(),
+            "expected layers not found in report",
+        );
+    };
+}
+
+#[test_log::test(tokio::test)]
+async fn report_roundtrip() -> Result<()> {
+    let reference = Reference::from_str("cgr.dev/chainguard/wolfi-base:latest")?;
```

Comment:
[nit] Depending on how often this changes it may be a good idea to select a single version, aren't the digests likely to change? 

I do definitely think there's a place for integration tests against moving targets which you're doing down below it seems.

Reply:
This is a fully synthetic round trip actually!